components/browser/state/src/main/java/mozilla/components/browser/state/reducer/DownloadStateReducer.kt

```diff
@@ -14,8 +15,9 @@ internal object DownloadStateReducer {
      */
     fun reduce(state: BrowserState, action: DownloadAction): BrowserState {
         return when (action) {
-            is DownloadAction.QueueDownloadAction -> {
-                state.copy(queuedDownloads = state.queuedDownloads + (action.download.id to action.download))
+            is DownloadAction.QueueDownloadAction -> updateQueuedDownloads(state, action.download)
+            is DownloadAction.UpdateQueuedDownloadAction -> {
```

Comment:
@csadilek  I couldn't use the same branch ( `is QueueDownloadAction, is UpdateQueuedDownloadAction ->`) as we will have to explicit cast  either to an `QueueDownloadAction` or to an `UpdateQueuedDownloadAction`. I preferred to use a shared function between them and use separated branches

Reply:
@Amejia481 yes, we'd have to introduce as shared base class. Probably not worth it.